sale_invoice_plan/tests/test_sale_invoice_plan.py

```diff
@@ -336,3 +284,56 @@ def test_03_unlink_invoice_plan(self):
         # Remove it
         self.so_service.remove_invoice_plan()
         self.assertFalse(self.so_service.invoice_plan_ids)
+
+    def test_invoice_plan_po_edit(self):
+        """Case when some installment already invoiced,
+        but then, the SO line added. Test to ensure that
+        the invoiced amount of the done installment is fixed"""
+        ctx = {
+            "active_id": self.so_service_so_edit.id,
+            "active_ids": [self.so_service_so_edit.id],
+            "all_remain_invoices": False,
+        }
+        f = Form(self.env["sale.create.invoice.plan"])
+        # Create Invoice Plan 5 installment
+        num_installment = 5
+        f.num_installment = num_installment
+        plan = f.save()
+        plan.with_context(**ctx).sale_create_invoice_plan()
+        # Change plan, so that the 1st installment is 1000 and 5th is 3000
+        self.assertEqual(len(self.so_service_so_edit.invoice_plan_ids), 5)
+        first_install = self.so_service_so_edit.invoice_plan_ids[0]
+        first_install.amount = 1000.0
+        self.so_service_so_edit.invoice_plan_ids[4].amount = 3000.0
+        self.so_service_so_edit.action_confirm()
+        self.assertEqual(self.so_service_so_edit.state, "sale")
+        sale_create = self.env["sale.make.planned.invoice"].create({})
+        # Create only the 1st invoice, amount should be 1000.0, and percent is 10
+        sale_create.with_context(**ctx).create_invoices_by_plan()
+        self.assertEqual(first_install.amount, 1000.0)
+        self.assertEqual(first_install.percent, 10)
+        # Add new SO line with amount = 1000, check that only percent is changed
+        self.so_service_so_edit.write(
+            {
+                "order_line": [
+                    (
+                        0,
+                        0,
+                        {
+                            "name": "SO-Product-NEW",
+                            "product_id": self.product_order.id,
+                            "product_uom_qty": 1,
+                            "product_uom": self.product_order.uom_id.id,
+                            "price_unit": 1000,
+                        },
+                    )
+                ],
+            }
+        )
+        # Overall amount changed to 11000, install amount not changed, only percent changed.
+        self.assertEqual(self.so_service_so_edit.amount_total, 11000)
+        self.so_service.invoice_plan_ids._compute_amount()
+        self.assertEqual(first_install.amount, 1000)
+        # Why is it 9.090909 here??? What is wrong for SO
+        # AssertionError: 10.0 != 9.090909
+        self.assertEqual(first_install.percent, 9.090909)
```

Comment:
hello @kittiu, can you pls help to resolve the test issue? I'm not an expert in this module.